From 7a3ddc4ceb64bf648a5f383814632f72b8700a8f Mon Sep 17 00:00:00 2001
From: James Page <james.page@ubuntu.com>
Date: Mon, 3 Sep 2018 11:40:32 +0100
Subject: [PATCH] py3: fix misc encoding issues

Fix miscellanous encoding issue when running under Python 3:

 - Encoding of member_id prior to passing into tooz.
 - Decoding of member ID's during response processing
   for status API calls.
---
 gnocchi/cli/metricd.py | 14 ++++++++------
 gnocchi/rest/api.py    |  6 ++++--
 gnocchi/rest/app.py    |  2 +-
 3 files changed, 13 insertions(+), 9 deletions(-)

diff --git a/gnocchi/cli/metricd.py b/gnocchi/cli/metricd.py
index 523c47a..3e8edcd 100644
--- a/gnocchi/cli/metricd.py
+++ b/gnocchi/cli/metricd.py
@@ -60,12 +60,14 @@ class MetricProcessBase(cotyledon.Service):
         self._wake_up.set()
 
     def _configure(self):
-        member_id = "%s.%s.%s" % (socket.gethostname(),
-                                  self.worker_id,
-                                  # NOTE(jd) Still use a uuid here so we're
-                                  # sure there's no conflict in case of
-                                  # crash/restart
-                                  str(uuid.uuid4()))
+        member_id = (
+            "%s.%s.%s" % (socket.gethostname(),
+                          self.worker_id,
+                          # NOTE(jd) Still use a uuid here so we're
+                          # sure there's no conflict in case of
+                          # crash/restart
+                          str(uuid.uuid4()))
+        ).encode()
         self.coord = get_coordinator_and_start(member_id,
                                                self.conf.coordination_url)
         self.store = storage.get_driver(self.conf)
diff --git a/gnocchi/rest/api.py b/gnocchi/rest/api.py
index 23d1fc3..02ddbaa 100644
--- a/gnocchi/rest/api.py
+++ b/gnocchi/rest/api.py
@@ -2106,9 +2106,11 @@ class StatusController(rest.RestController):
                     metricd.MetricProcessor.GROUP_ID, member)
                 for member in members
             ]
-            report_dict['metricd']['processors'] = members
+            report_dict['metricd']['processors'] = [
+                member.decode() for member in members
+            ]
             report_dict['metricd']['statistics'] = {
-                member: cap.get()
+                member.decode(): cap.get()
                 for member, cap in six.moves.zip(members, caps)
             }
         else:
diff --git a/gnocchi/rest/app.py b/gnocchi/rest/app.py
index 9838466..c4a6f89 100644
--- a/gnocchi/rest/app.py
+++ b/gnocchi/rest/app.py
@@ -102,7 +102,7 @@ class GnocchiHook(pecan.hooks.PecanHook):
                         # entirely.
                         self.backends[name] = (
                             metricd.get_coordinator_and_start(
-                                str(uuid.uuid4()),
+                                str(uuid.uuid4()).encode(),
                                 self.conf.coordination_url)
                         )
                     elif name == "storage":
-- 
2.7.4

